hw/ip/otbn/dv/otbnsim/sim/csr.py

```diff
@@ -15,17 +11,32 @@ class CSRFile:
     def __init__(self) -> None:
         self.flags = FlagGroups()
 
+    @staticmethod
+    def _read_field(field_idx: int, field_size: int, val: int) -> int:
```

Comment:
Just realised there's probably a better way to do this, via the `__getitem__` and `__setitem__` on the flags object, will redo with that.

Reply:
I'd probably prefer the more explicit version, like you've got already. Overloading `__getitem__` / `__setitem__` conflicts slightly with the "blindingly obvious what it does" goal in the ISS.

Reply:
Though could leave these all the same for use with mod? Maybe pointless to introduce them purely for that but may give neater code, thoughts?

Reply:
> I'd probably prefer the more explicit version

Suits me, less rework required for me :)

Reply:
Oh, also maybe these would be better as `get_field` and `set_field` since they don't actually read or write anything. In particular, I initially assumed that calling `_write_field` would cause a write.